crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -75,42 +76,64 @@ impl OperationLogs {
+
+        // FIXME: this is a hotfix to map "software-list" and "software-update" to "software-management"
+        // this should be fixed in https://github.com/thin-edge/thin-edge.io/issues/1077
         for file in (self.log_dir.read_dir()?).flatten() {
             if let Some(path) = file.path().file_name().and_then(|name| name.to_str()) {
-                if path.starts_with(UPDATE_PREFIX) {
-                    // The paths are pushed using the reverse alphabetic order
-                    update_logs.push(Reverse(path.to_string()));
-                } else if path.starts_with(LIST_PREFIX) {
-                    // The paths are pushed using the reverse alphabetic order
-                    list_logs.push(Reverse(path.to_string()));
+                if path.starts_with("software-list") {
+                    log_tracker
+                        .entry("software-list".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else if path.starts_with("software-update") {
+                    log_tracker
+                        .entry("software-update".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else {
+                    let file_name = path
+                        .split('-')
+                        .next()
+                        .ok_or(OperationLogsError::FileFormatError)?;
+                    log_tracker
+                        .entry(file_name.to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
                 }
             }
         }
 
-        while update_logs.len() > self.max_update_logs {
-            if let Some(rname) = update_logs.pop() {
-                let name = rname.0;
-                let path = self.log_dir.join(name.clone());
-                if let Err(err) = std::fs::remove_file(&path) {
-                    log::warn!("Fail to remove out-dated log file {} : {}", name, err);
-                }
+        for (key, value) in log_tracker.iter_mut() {
+            if key.starts_with("software-list") {
+                // only allow one update list file in logs
+                let () = remove_n_logs(value, &self.log_dir, 1)?;
+            } else {
+                // allow most recent five
+                let () = remove_n_logs(value, &self.log_dir, 5)?;
             }
         }
 
-        while list_logs.len() > self.max_list_logs {
-            if let Some(rname) = list_logs.pop() {
-                let name = rname.0;
-                let path = self.log_dir.join(name.clone());
-                if let Err(err) = std::fs::remove_file(&path) {
-                    log::warn!("Fail to remove out-dated log file {} : {}", name, err);
-                }
+        Ok(())
+    }
+}
+
+fn remove_n_logs(
```

Comment:
Minor - This name sounds as if `n` log files were removed. Why not something like `remove_old_logs`?